proj4/src/main/java/org/osgeo/proj4j/BasicCoordinateTransform.java

```diff
@@ -156,11 +160,20 @@ private void datumTransform(ProjCoordinate pt)
         /* -------------------------------------------------------------------- */
         /*      Short cut if the datums are identical.                          */
         /* -------------------------------------------------------------------- */
-        if (srcCRS.getDatum().isEqual(tgtCRS.getDatum()))
+        if (srcCRS.getDatum().isEqual(tgtCRS.getDatum())
+                || srcCRS.getDatum().getTransformType() == Datum.TYPE_UNKNOWN
+                || tgtCRS.getDatum().getTransformType() == Datum.TYPE_UNKNOWN)
             return;
     
-        // TODO: grid shift if required
+        if (srcCRS.getDatum().getTransformType() == Datum.TYPE_GRIDSHIFT) {
+            srcCRS.getDatum().shift(pt);
+            srcGeoConv.overrideWithWGS84Params();

```

Comment:
Override WGS84 params in constructor of basic coordinate transform, instead of doing it per point transformed.